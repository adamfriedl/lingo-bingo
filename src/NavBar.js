// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import { Absolute, Toolbar, NavLink } from 'rebass';

const NavBar = (props: { color: string }) => {
  return (
    <div>
      <Absolute z={2} top left>
        <Toolbar w={'100%'} bg="transparent" color={props.color}>
          <Link to={`${process.env.PUBLIC_URL}/`}>
            <NavLink children="Home" />
          </Link>

          <Link to={`${process.env.PUBLIC_URL}/about`}>
            <NavLink children="About" />
          </Link>
        </Toolbar>
      </Absolute>
    </div>
  );
};
export default NavBar;
