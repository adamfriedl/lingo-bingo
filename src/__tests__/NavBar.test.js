import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../NavBar';

it('renders', () => {
  const component = shallow(<NavBar />);
  expect(component).toMatchSnapshot();
});
