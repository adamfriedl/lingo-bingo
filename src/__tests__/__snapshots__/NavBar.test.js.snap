// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Absolute
        left={true}
        right={true}
        top={true}
        z={2}
>
        <Toolbar
                bg="transparent"
                color="teal3"
                pl={2}
                pr={2}
                w="100%"
        >
                <NavLink
                        color="inherit"
                        f={1}
                        p={2}
                >
                        <Styled(Link)
                                to="/"
                        >
                                Home
                        </Styled(Link)>
                </NavLink>
                <NavLink
                        color="inherit"
                        f={1}
                        p={2}
                >
                        <Styled(Link)
                                to="/about"
                        >
                                About
                        </Styled(Link)>
                </NavLink>
                <NavLink
                        color="inherit"
                        f={1}
                        href="http://adamfriedl.com"
                        ml="auto"
                        mr={
                                Array [
                                        2,
                                        3,
                                      ]
                        }
                        p={2}
                        style={
                                Object {
                                        "fontFamily": "Damion",
                                      }
                        }
                >
                        by Adam Friedl
                </NavLink>
        </Toolbar>
</Absolute>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Toolbar
          bg="transparent"
          color="teal3"
          pl={2}
          pr={2}
          w="100%"
>
          <NavLink
                    color="inherit"
                    f={1}
                    p={2}
          >
                    <Styled(Link)
                              to="/"
                    >
                              Home
                    </Styled(Link)>
          </NavLink>
          <NavLink
                    color="inherit"
                    f={1}
                    p={2}
          >
                    <Styled(Link)
                              to="/about"
                    >
                              About
                    </Styled(Link)>
          </NavLink>
          <NavLink
                    color="inherit"
                    f={1}
                    href="http://adamfriedl.com"
                    ml="auto"
                    mr={
                              Array [
                                        2,
                                        3,
                                      ]
                    }
                    p={2}
                    style={
                              Object {
                                        "fontFamily": "Damion",
                                      }
                    }
          >
                    by Adam Friedl
          </NavLink>
</Toolbar>,
        "left": true,
        "right": true,
        "top": true,
        "z": 2,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bg": "transparent",
          "children": Array [
            <NavLink
              color="inherit"
              f={1}
              p={2}
>
              <Styled(Link)
                            to="/"
              >
                            Home
              </Styled(Link)>
</NavLink>,
            <NavLink
              color="inherit"
              f={1}
              p={2}
>
              <Styled(Link)
                            to="/about"
              >
                            About
              </Styled(Link)>
</NavLink>,
            <NavLink
              color="inherit"
              f={1}
              href="http://adamfriedl.com"
              ml="auto"
              mr={
                            Array [
                                          2,
                                          3,
                                        ]
              }
              p={2}
              style={
                            Object {
                                          "fontFamily": "Damion",
                                        }
              }
>
              by Adam Friedl
</NavLink>,
          ],
          "color": "teal3",
          "pl": 2,
          "pr": 2,
          "w": "100%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Link)
                to="/"
>
                Home
</Styled(Link)>,
              "color": "inherit",
              "f": 1,
              "p": 2,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Link)
                to="/about"
>
                About
</Styled(Link)>,
              "color": "inherit",
              "f": 1,
              "p": 2,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "About",
                "to": "/about",
              },
              "ref": null,
              "rendered": "About",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "by Adam Friedl",
              "color": "inherit",
              "f": 1,
              "href": "http://adamfriedl.com",
              "ml": "auto",
              "mr": Array [
                2,
                3,
              ],
              "p": 2,
              "style": Object {
                "fontFamily": "Damion",
              },
            },
            "ref": null,
            "rendered": "by Adam Friedl",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Absolute
          left={true}
          right={true}
          top={true}
          z={2}
>
          <Toolbar
                    bg="transparent"
                    color="teal3"
                    pl={2}
                    pr={2}
                    w="100%"
          >
                    <NavLink
                              color="inherit"
                              f={1}
                              p={2}
                    >
                              <Styled(Link)
                                        to="/"
                              >
                                        Home
                              </Styled(Link)>
                    </NavLink>
                    <NavLink
                              color="inherit"
                              f={1}
                              p={2}
                    >
                              <Styled(Link)
                                        to="/about"
                              >
                                        About
                              </Styled(Link)>
                    </NavLink>
                    <NavLink
                              color="inherit"
                              f={1}
                              href="http://adamfriedl.com"
                              ml="auto"
                              mr={
                                        Array [
                                                  2,
                                                  3,
                                                ]
                              }
                              p={2}
                              style={
                                        Object {
                                                  "fontFamily": "Damion",
                                                }
                              }
                    >
                              by Adam Friedl
                    </NavLink>
          </Toolbar>
</Absolute>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Toolbar
            bg="transparent"
            color="teal3"
            pl={2}
            pr={2}
            w="100%"
>
            <NavLink
                        color="inherit"
                        f={1}
                        p={2}
            >
                        <Styled(Link)
                                    to="/"
                        >
                                    Home
                        </Styled(Link)>
            </NavLink>
            <NavLink
                        color="inherit"
                        f={1}
                        p={2}
            >
                        <Styled(Link)
                                    to="/about"
                        >
                                    About
                        </Styled(Link)>
            </NavLink>
            <NavLink
                        color="inherit"
                        f={1}
                        href="http://adamfriedl.com"
                        ml="auto"
                        mr={
                                    Array [
                                                2,
                                                3,
                                              ]
                        }
                        p={2}
                        style={
                                    Object {
                                                "fontFamily": "Damion",
                                              }
                        }
            >
                        by Adam Friedl
            </NavLink>
</Toolbar>,
          "left": true,
          "right": true,
          "top": true,
          "z": 2,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bg": "transparent",
            "children": Array [
              <NavLink
                color="inherit"
                f={1}
                p={2}
>
                <Styled(Link)
                                to="/"
                >
                                Home
                </Styled(Link)>
</NavLink>,
              <NavLink
                color="inherit"
                f={1}
                p={2}
>
                <Styled(Link)
                                to="/about"
                >
                                About
                </Styled(Link)>
</NavLink>,
              <NavLink
                color="inherit"
                f={1}
                href="http://adamfriedl.com"
                ml="auto"
                mr={
                                Array [
                                                2,
                                                3,
                                              ]
                }
                p={2}
                style={
                                Object {
                                                "fontFamily": "Damion",
                                              }
                }
>
                by Adam Friedl
</NavLink>,
            ],
            "color": "teal3",
            "pl": 2,
            "pr": 2,
            "w": "100%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Link)
                  to="/"
>
                  Home
</Styled(Link)>,
                "color": "inherit",
                "f": 1,
                "p": 2,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Link)
                  to="/about"
>
                  About
</Styled(Link)>,
                "color": "inherit",
                "f": 1,
                "p": 2,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "About",
                  "to": "/about",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "by Adam Friedl",
                "color": "inherit",
                "f": 1,
                "href": "http://adamfriedl.com",
                "ml": "auto",
                "mr": Array [
                  2,
                  3,
                ],
                "p": 2,
                "style": Object {
                  "fontFamily": "Damion",
                },
              },
              "ref": null,
              "rendered": "by Adam Friedl",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
